core/templates/filters/translate.pipe.ts

```diff
@@ -0,0 +1,133 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Translate pipe for i18n translations.
+ */
+
+import { ChangeDetectorRef, OnDestroy, Pipe, PipeTransform }
+  from '@angular/core';
+import { Subscription } from 'rxjs';
+import { LangChangeEvent, TranslateService } from 'services/translate.service';
+import { UtilsService } from 'services/utils.service';
+
+/**
+ * Commonly used terms in this file.
+ * Note: Intentionally left out the L of innerHTM"L" to avoid the linting error.
+ * Example: <h1 [innerHTM]="'I18N_ABOUT_PAGE_HEADING' | translate:{x: 'val'}">
+ * 'I18N_ABOUT_PAGE_HEADING' is referred here as key.
+ * "translate" is the pipe. Every pipe must have a transform function. The
+ * transform function is called when angular encouters the pipe in HTML.
+ * The object following the pipe, i.e.{x: 'val'}, is another argument to the
+ * transform function. This object is called params or interpolationParams.
+ */
+
+@Pipe({
+  name: 'translate',
+  // This option is required to update the value when promises are resolved.
+  pure: false
+})
+export class TranslatePipe implements PipeTransform, OnDestroy {
+  interpolatedValue: string = '';
+  lastKey: string;
```

Comment:
Don't know what lastKey is. Maybe add single-line comments above each of these with a description of what they mean/represent.

Reply:
Added